src/main/java/com/olgaboiar/mint/RequestParser.java

```diff
@@ -1,17 +1,49 @@
 package com.olgaboiar.mint;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public class RequestParser {
-    Request parse(List<String> incomingRequest) {
-        String[] requestHeader = incomingRequest.get(0).split(" ");
-        String method = requestHeader[0];
-        String requestedFile = requestHeader[1];
-        String protocol = requestHeader[2];
-        Request parsedRequest = new Request();
-        parsedRequest.setMethod(method);
-        parsedRequest.setRequestedFile(requestedFile);
-        parsedRequest.setProtocol(protocol);
+    Request parse(List<String> incomingRequest) throws MalformedURLException {
+        String[] requestLine = incomingRequest.get(0).split(" ");
+        Map<String, String> requestHeaders = parseRequestHeaders(incomingRequest);
+        URL url = new URL(parseProtocol(requestLine), parseHost(requestHeaders), parsePort(requestHeaders), parsePath(requestLine));
+        Request parsedRequest = new Request(url, parseMethod(requestLine));
         return parsedRequest;
     }
+
+    private Map<String, String> parseRequestHeaders(List<String> incomingRequest) {
+        List<String> headers = incomingRequest.subList(1, incomingRequest.size());
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        String[] restArr = new String[headers.size()];
+        restArr = headers.toArray(restArr);
+        for (int i = 0; i < headers.size(); i ++) {
+            String[] line = restArr[i].split(": ");
+            requestHeaders.put(line[0], line[1]);
+        }
+        return requestHeaders;
+    }
+
+    private Integer parsePort(Map<String, String> requestHeaders) {
+        return Integer.parseInt(requestHeaders.get("Host").split(":")[1]);
+    }
+
+    private String parseHost(Map<String, String> requestHeaders) {
+        return requestHeaders.get("Host").split(":")[0];
+    }
+
+    private String parseProtocol(String[] requestLine) {
+        return requestLine[2].split("/")[0].toLowerCase();
+    }
+
+    private String parseMethod(String[] requestLine) {
+        return requestLine[0];
+    }
+
+    private String parsePath(String[] requestLine) {
+        return requestLine[1];
+    }
```

Comment:
As discussed, line 18-48 is a potential target for extract class.

Reply:
fixed in https://github.com/olgaboiar/MintHTTPServer/pull/9/commits/f195c9c37703361437af0136ebaa3ab904044533

Reply:
extracted in https://github.com/olgaboiar/MintHTTPServer/pull/10/commits/e55cb313b9f578e8479e3fd4719db3209331f057